src/parameter.cpp

```diff
@@ -132,6 +132,11 @@ parameter::operator uint64_t() const
         return intValue;
     }
 
+    static constexpr long double uint64_max = std::numeric_limits<uint64_t>::max();
+    if (type == DDWAF_OBJ_FLOAT && (f64 >= 0.0) && (f64 <= uint64_max)) {
+        return static_cast<uint64_t>(f64);
+    }
```

Comment:
I'd add `f64 == round(f64)` or `f64 == static_cast<uint64_t>(f64)` to avoid truncation of the decimal part.

Reply:
I can add this to both but I was trying to be extra lenient...